### PR TITLE
wire: improve error for a provider set with a binding but no corresponding provider

### LINKED ISSUE #113: wire: improve error message when a provider set includes a binding but not the corresponding concrete type
### Describe the bug

A wire set containing __subsets__ (one or more wire sets), should be just the same as a wire set with items from that subsets added directly.

### To Reproduce

Given this code:

__master.go__
```go
package wirebug

type ISomeInterface interface {
	DoStuff(thing string) int
}

type SomeStuff struct{}

func (si *SomeStuff) DoStuff(thing string) int {
	return len(thing)
}

func NewStuff() *SomeStuff {
	return &SomeStuff{}
}
```

__wiring.go__
```go
package wirebug

import "github.com/google/wire"

var WiringProviders = wire.NewSet(
	NewStuff,
)

var WiringBindings = wire.NewSet(
	wire.Bind(new(ISomeInterface), new(SomeStuff)),
)

var WiringSet = wire.NewSet(
	WiringProviders,
	WiringBindings,
)
```

__injectors.go__
```go
//+build wireinject

package wirebug

import "github.com/google/wire"

func InjectStuff() ISomeInterface {
	wire.Build(WiringSet)
	return nil
}
```

Running `wire ./...` results in an error:
```
$$$/wirebug/wiring.go:10:2: no binding for *$$$/wirebug.SomeStuff
wire: at least one generate failure
```
It's as if items from `WiringProviders` can't be seen by `WiringBindings` although they're inside `WiringSet`. Inside the documentation of `wire.NewSet`, there's this statement, **"Passing a ProviderSet to NewSet is the same as if the set's contents were passed as arguments to NewSet directly."**, but it turns out that it's not so.

## Expected behavior

The file `wire_gen.go` should be generated without problem.

Both of these `WiringSet`
```go
var WiringProviders = wire.NewSet(
	NewStuff,
)

var WiringBindings = wire.NewSet(
	wire.Bind(new(ISomeInterface), new(SomeStuff)),
)

var WiringSet = wire.NewSet(
	WiringProviders,
	WiringBindings,
)
```
and
```go
var WiringSet = wire.NewSet(
	NewStuff,
	wire.Bind(new(ISomeInterface), new(SomeStuff)),
)
```
should behave the same. But only the below version works.

### Version

Version 0.2.1

### Additional context

None.

### PR DESCRIPTION
**Pull request by vangent**
Fixes #113.

### CODE REVIEW

File: internal/wire/analyze.go [hunk 1]
```diff
@@ -353,9 +353,11 @@ func buildProviderMap(fset *token.FileSet, hasher typeutil.Hasher, set *Provider
 		}
 		concrete := providerMap.At(b.Provided)
 		if concrete == nil {
-			pos := fset.Position(b.Pos)
-			typ := types.TypeString(b.Provided, nil)
-			ec.add(notePosition(pos, fmt.Errorf("no binding for %s", typ)))
+			setName := set.VarName
+			if setName == "" {
+				setName = "wire.Build"
```
Comment (zombiezen) [inline internal/wire/analyze.go]: This isn't always the case: one can always write an anonymous `wire.NewSet` call in the middle of a larger expression.

Comment (vangent) [reply internal/wire/analyze.go]: Hmm, I copied this from somewhere else (`bindingConflictError`, near the end of file). What do you suggest? "Anonymous wire.NewSet"?

Comment (zombiezen) [reply internal/wire/analyze.go]: Perhaps just "provider set"? The line number should already convey where the problem is.

Comment (vangent) [reply internal/wire/analyze.go]: Done.

**Pull request merged**